src/modules/shared/components/GenericButtonActions/GenericButtonActions.tsx

```diff
@@ -13,15 +12,21 @@ interface ButtonProps {
 
 export const GenericButtonActions = ({ dataItem, actions }: ButtonProps) => {
   const [isMenuOpen, setIsMenuOpen] = useState(false);
-  const router = useRouter();
 
   const renderOptions = useMemo(() => {
     return actions.filter((item) =>
       item.conditions ? item.conditions(dataItem) : true
     );
   }, [actions, dataItem]);
 
-  const handleAction = (action: any) => {
+  const handleAction = (event: any, action: any) => {
+    event?.preventDefault();
```

Comment:
Só deve chamar o preventDefault se for do tipo function. Se não ele não vai deixar o href funcionar.